src/block/block_ext.c

```diff
@@ -31,6 +31,13 @@ static int __block_merge(WT_SESSION_IMPL *, WT_BLOCK *, WT_EXTLIST *, wt_off_t,
 /*
  * __block_off_srch_last --
  *     Return the last element in the list, along with a stack for appending.
+ *
+ * If head[i] == NULL then stack[i] = &head[i]. Thus setting *stack[i] in __block_append updates
+ *     WT_EXTLIST.off[i] to point to the new last extent since __block_off_srch_last is called with
+ *     head = WT_EXTLIST.off[].
+ *
+ * If head[i] != NULL stack[i] is not changed. Thus setting *stack[i] in __block_append updates
+ *     WT_EXT.next[] of the former last extent to point to the new last extent.
```

Comment:
I find it more useful to describe the behavior of a function at a higher level. A developer can read the code to determine precisely how the elements of `stack` get set.

Maybe something like:

> Return a stack such that the caller can append a new entry to the skip list by inserting it after each element in the stack. For non-empty levels, this will be the last element at that level of the skip list. For a level with no entries, this will be the corresponding entry in the head stack.

Reply:
Done.